src/net/transport.rs

```diff
@@ -153,16 +153,12 @@ impl Transport for Arc<HttpTransport> {
                 let step =
                     <Option<Gate>>::from(route.gate()).expect("step required when sending records");
                 let resp_future = self.clients[dest].step(query_id, &step, data)?;
-                tokio::spawn(async move {
```

Comment:
@andyleiserson not sure why we had to spawn a task here, but all tests pass when I removed it. That turned out to be beneficial as `resp_future` no longer need to be `'static`

Reply:
I can't remember either, seems reasonable to remove.